kernel/classes/ezcontentobjecttreenode.php

```diff
@@ -2842,15 +2823,15 @@ static function assignSectionToSubTree( $nodeID, $sectionID, $oldSectionID = fal
         }
 
         $db->begin();
-        foreach ( array_chunk( $objectSimpleIDArray, 100 ) as $pagedObjectIDs )
+        foreach ( array_chunk( $objectIDArray, 100 ) as $pagedObjectIDs )
         {
             $db->query( "UPDATE ezcontentobject SET section_id='$sectionID' WHERE $filterPart " . $db->generateSQLINStatement( $pagedObjectIDs, 'id', false, true, 'int' ) );
             eZSearch::updateObjectsSection( $pagedObjectIDs, $sectionID );
         }
         $db->commit();
 
         // clear caches for updated objects
-        eZContentObject::clearCache( $objectSimpleIDArray );
+        eZContentObject::clearCache( $objectIDArray );
```

Comment:
I'm still not fully able to see what is changed here besides the variable name (and the two other new issues written above)

Could you expand on what you are fixing in this part of the change _(ezcontentobjecttreenode.php)_?

Reply:
Sure: in the original code, there is a part where [$objectSimpleIDArray](https://github.com/ezsystems/ezpublish-legacy/blob/master/kernel/classes/ezcontentobjecttreenode.php#L2831) is defined only to store id values from another array [$objectIDArray](https://github.com/ezsystems/ezpublish-legacy/blob/master/kernel/classes/ezcontentobjecttreenode.php#L2811) which does not contain objects IDs but whole records returned from DB - this is a case of bad variable naming. The code responsible for getting records from DB and creating the array of IDs for it was moved to the `getObjectIdsInNodeSubTree` method, so there is no need for little misleading `$objectSimpleIDArray` variable.

To summarize: the code responsible for `$objectSimpleIDArray` was moved to another method thus this variable (variable name) is no longer needed.

Reply:
So we don't really need to risk changing this part of the code to solve the bug right? ;)